src/dfx/src/lib/operations/cycles_ledger.rs

```diff
@@ -16,16 +16,16 @@ use slog::{info, Logger};
 const ICRC1_BALANCE_OF_METHOD: &str = "icrc1_balance_of";
 const ICRC1_TRANSFER_METHOD: &str = "icrc1_transfer";
 const SEND_METHOD: &str = "send";
+const CYCLES_LEDGER_CANISTER_ID: &str = "um5iw-rqaaa-aaaaq-qaaba-cai";
```

Comment:
Would you please handle this the same way we do MAINNET_CYCLE_MINTER_CANISTER_ID?  That way it's a `const Principal`, and doesn't have to be converted from a string.

Reply:
I thought about it, but I was concerned about the fact that the canister ID becomes effectively almost non-searchable:
```rust
pub const MAINNET_CYCLE_MINTER_CANISTER_ID: Principal =
    Principal::from_slice(&[0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x04, 0x01, 0x01]);
```
, meaning, if I only knew the ID of the minter canister `mqygn-kiaaa-aaaar-qaadq-cai` (which can be found on forum, for example) and would like to search through DFINITY GitHub org to find all references in the code, I wouldn't find it. I also concluded the time to compute Principal from String should be negligible from the user's perspective. 

But there may be other good reasons to store it as a hex which I don't know about, and if so, then I'll annotate with a comment with the canister ID above.

Reply:
Part of the approach involves a test that verifies that the const Principal matches the expected value. See https://github.com/dfinity/sdk/blob/master/src/dfx/src/lib/ledger_types/mod.rs#L181 for an example with MAINNET_CYCLE_MINTER_CANISTER_ID.

It's not about execution time at all (which should be obvious).  It's about having a `const Principal` that doesn't need to go through .into() and can be passed to any method that takes &Principal, and also about consistency.  We shouldn't do it one way for MAINNET_LEDGER_CANISTER_ID, MAINNET_CYCLE_MINTER_CANISTER_ID, MAINNET_MOTOKO_PLAYGROUND_CANISTER_ID, DANK_PRINCIPAL, MAINNET_BITCOIN_CANISTER_ID and another way for CYCLES_LEDGER_CANISTER_ID.  (That said, there are two others that snuck by with const &str).